frontend/src/pages/DataSources/DataSources.tsx

```diff
@@ -23,65 +23,87 @@ interface ReduxState {
   sources: SourcesState;
   page: DataSourcesState;
 }
+
+export interface DataSource {
+  onMetadataClick?: (source: SourceMap) => void;
+  onDatasetClick?: (source: SourceMap, sourceId: number) => void;
+}
 type DataSourcesProps = ReduxState & ActionProps & RouteComponentProps;
 
-class DataSources extends React.Component<DataSourcesProps> {
-  render(): ReactElement {
-    const sources = this.props.sources.get('sources') as List<SourceMap>;
-    const loading = this.props.sources.get('loading') as boolean;
-    const activeSource = this.props.page.get('activeSource') as SourceMap | undefined;
+export const DataSourcesContext = createContext<DataSource>({});
+
+const DataSources: FunctionComponent<DataSourcesProps> = (props) => {
+  const sources = props.sources.get('sources') as List<SourceMap>;
+  const loading = props.sources.get('loading') as boolean;
+  const activeSource = props.page.get('activeSource') as SourceMap | undefined;
+  const [showModal, setShowModal] = useState(false);
+  const history = useHistory();
+
+  const renderDetailsTab = (activeSource: SourceMap | undefined, loading = false): ReactNode => {
+    if (activeSource && !loading && showModal) {
+      return (
+        <Modal show={showModal} onHide={hideModal}>
+          <SourceDetailsTab source={activeSource} />
+        </Modal>
+      );
+    }
 
     return (
+      <Segment>
+        <Placeholder>
+          <Placeholder.Header>
+            <Placeholder.Line length="very short" />
+            <Placeholder.Line length="medium" />
+          </Placeholder.Header>
+          <Placeholder.Paragraph>
+            <Placeholder.Line length="short" />
+          </Placeholder.Paragraph>
+        </Placeholder>
+      </Segment>
+    );
```

Comment:
Do you know what these are for? I don't think they are required with the current setup.